client/shipping-label/state/actions.js

```diff
@@ -452,14 +450,19 @@ export const purchaseLabel = () => ( dispatch, getState, context ) => {
 				console.error( error );
 				dispatch( NoticeActions.errorNotice( error.toString() ) );
 			} else {
-				const purchasedLabels = _.differenceBy( response, oldLabels, 'label_id' );
-				const labelsToPrint = purchasedLabels.map( ( label, index ) => ( {
-					caption: sprintf( __( 'PACKAGE %d (OF %d)' ), index + 1, purchasedLabels.length ),
+				const labelsToPrint = response.map( ( label, index ) => ( {
+					caption: sprintf( __( 'PACKAGE %d (OF %d)' ), index + 1, response.length ),
 					labelId: label.label_id,
 				} ) );
 				const state = getState().shippingLabel;
 				printDocument( getPrintURL( state.paperSize, labelsToPrint, context ) )
-					.then( () => dispatch( exitPrintingFlow() ) )
+					.then( () => {
+						const noticeText = 1 === response.length
+								? __( 'Your shipping label was purchased successfully' )
+								: sprintf( __( 'Your %d shipping labels were purchased successfully' ), response.length );
+						dispatch( NoticeActions.successNotice( noticeText ) );
```

Comment:
I've ended up implementing the notice suggested by #792 . It doesn't go away after any time, but it's dismissable.